sdk/vppagent/acl.go

```diff
@@ -0,0 +1,319 @@
+}
+
+// Request implements the request handler
+func (a *ACL) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*connection.Connection, error) {
+
+	if a.GetNext() == nil {
+		logrus.Fatal("The VPP Agent ACL composite requires that there is Next set")
+	}
+
+	incomingConnection, err := a.GetNext().Request(ctx, request)
+	if err != nil {
+		logrus.Errorf("Next request failed: %v", err)
+		return nil, err
+	}
+
+	opaque := a.GetNext().GetOpaque(incomingConnection)
+	if opaque == nil {
+		err = fmt.Errorf("received empty data from Next")
+		logrus.Errorf("Unable to find connection data: %v", err)
+		return nil, err
+	}
+	connectionData := opaque.(*ConnectionData)
+
+	if connectionData.SrcName == "" {
+		err = fmt.Errorf("found empty source name")
+		logrus.Errorf("Invalid connection data: %v", err)
+		return nil, err
+	}
+
+	connectionData.DataChange, err = a.appendDataChange(connectionData.DataChange, connectionData.SrcName)
+	if err != nil {
+		logrus.Error(err)
+		return nil, err
+	}
+
+	a.Connections[incomingConnection.GetId()] = connectionData
+
+	return incomingConnection, nil
+}
+
+// Close implements the close handler
+func (a *ACL) Close(ctx context.Context, connection *connection.Connection) (*empty.Empty, error) {
+	if a.GetNext() != nil {
+		return a.GetNext().Close(ctx, connection)
+	}
+	return &empty.Empty{}, nil
+}
+
+// GetOpaque will return the corresponding connection data
+func (a *ACL) GetOpaque(incoming interface{}) interface{} {
+	incomingConnection := incoming.(*connection.Connection)
+	if connectionData, ok := a.Connections[incomingConnection.GetId()]; ok {
+		return connectionData
+	}
+	logrus.Errorf("GetOpaque outgoing not found for %v", incomingConnection)
+	return nil
+}
+
+// Name returns the composite name
+func (a *ACL) Name() string {
+	return "acl"
+}
+
+// NewACL creates an ACL
+func NewACL(configuration *common.NSConfiguration, rules map[string]string) *ACL {
+	// ensure the env variables are processed
+	if configuration == nil {
+		configuration = &common.NSConfiguration{}
+	}
+	configuration.CompleteNSConfiguration()
+
+	return &ACL{
+		Rules:       rules,
+		Connections: map[string]*ConnectionData{},
+	}
+}
+
+func (a *ACL) appendDataChange(rv *configurator.Config, ingressInterface string) (*configurator.Config, error) {
+	if rv == nil {
+		rv = &configurator.Config{}
+	}
+	if rv.VppConfig == nil {
+		rv.VppConfig = &vpp.ConfigData{}
+	}
+
+	rules := []*acl.ACL_Rule{}
+
+	for _, rule := range a.Rules {
+		parsed := tools.ParseKVStringToMap(rule, ",", "=")
+
+		action, err := getAction(parsed)
+		if err != nil {
+			logrus.Errorf("Parsing rule %s failed with %v", rule, err)
+			return nil, err
+		}
+
+		match, err := getMatch(parsed)
+		match.Action = action
+		if err != nil {
+			logrus.Errorf("Parsing rule %s failed with %v", rule, err)
+			return nil, err
+		}
+
+		rules = append(rules, match)
+
+		rv.VppConfig.Acls = append(rv.VppConfig.Acls, &acl.ACL{
+			Name:  "IngressACL",
```

Comment:
Should names to be taken from somewhere? from rule's itself? It should not be same I suppose.

Reply:
@nickolaev could you please comment on this?

Reply:
Maybe we can append a name of an interface to make it kinda unique

Reply:
Do not find in the doc anything special about ACL name, seems it should be unique only. Added interface name into it.